src/dispatch/plugins/dispatch_jira/plugin.py

```diff
@@ -258,24 +258,27 @@ def create(
         """Creates an incident Jira issue."""
         client = create_client(self.configuration)
 
-        assignee = get_user_field(
-            client, self.configuration.hosting_type, self.configuration.username, commander_email
-        )
-        reporter = get_user_field(
-            client, self.configuration.hosting_type, self.configuration.username, reporter_email
-        )
+        assignee = get_user_field(client, self.configuration, commander_email)
+        reporter = get_user_field(client, self.configuration, reporter_email)
 
         project_id, issue_type_name = process_plugin_metadata(incident_type_plugin_metadata)
 
         if not project_id:
             project_id = self.configuration.default_project_id
 
+        # NOTE: to support issue creation by project id or key
+        project = {"id": project_id}
+        if not isinstance(project_id, int):
+            project = {"key": project_id}
```

Comment:
This is to support both Jira project ids and keys in the configuration fields, as on-prem ids do not translate to cloud and it's easier to work with keys.